src/controller/user-controller.ts

```diff
@@ -778,13 +768,8 @@ export default class UserController extends BaseController {
         return;
       }
 
-      user = {
-        ...body,
-      } as User;
-      await User.update(parameters.id, user);
-      res.status(200).json(
-        await UserService.getSingleUser(asNumber(parameters.id)),
-      );
+      const response = await UserService.updateUser(id, req.body);
```

Comment:
This won't work that easily. 

Going from a GEWIS Member to a Local User should also remove the GEWIS member info right? The LDAP integration is also based on GEWIS m-numbers.

In general there is quite some tricky stuff, since there are 2 ways accounts are created in SudoSOS: 
- A new user is created by logging in using the GEWIS website (and the account is bound to the m-number)
- A new user is created when logging in using LDAP and no account associated with the m-number exists.

This means that removing someones MEMBER type status MUST only be done when you are sure they don't have a GEWIS account anymore. 

A solution could be to keep the GEWIS m-number info stored, but just throw an error if someones tries to log in based on m-number whilst its not a type MEMBER? (I _believe_ this is almost the current behavior). 

@tomudding would it be bad if we kept the m-number to SudoSOS id info stored?

Reply:
> @tomudding would it be bad if we kept the m-number to SudoSOS id info stored?

Ideally, we never keep it to prevent people from re-associating the transaction data to the user once we start deleting data (i.e. the yet to be implemented anonymisation process - though note that people can technically already request this using the GDPR).

However, at this point, we are not yet deleting the data but only changing types, as such, I think it is fine to keep track of the user's m-number if it exists. Preventing token creation/login when trying to use a MEMBER style login is also logical, but you will have to make sure that this is properly documented within the codebase (and properly handle it in both the dashboard and POS)!

Reply:
So what do we want to do? Because the workflow seems much more complex than I imagined. I personally don't think we should make a dependency on the GEWIS database to verify whether MEMBER accounts are actually deleted from the GEWIS database, because it seems unnecessary? I also think that keeping m-account related information might not cause any harm, because that would make changing the UserType a revertible change, and a not a destructive one.

Reply:
Keep the data, for now, but maybe add the following:

The main use case for switching UserType is from MEMBER to LOCAL, however a LOCAL user needs a password. So I say we send the person an info mail with a (re)set password link? And maybe add some integration test cases that a new account wont be created in both the cases when the m-number still exists but instead errors?

This might be a bit more involved, so let me know if you still want to do this.
When I looked at it, just adding an extra check to the endpoints in the `gewis-authentication-controller` [here](https://github.com/GEWIS/sudosos-backend/blob/e8cf1d272e989ca62463f2f48211c7ba5758c35c/src/gewis/controller/gewis-authentication-controller.ts#L118) might be a solid move, since that would throw a nice error. (return a 403, error m-number unregistered).

As a possible extra you could also throw an error [here](https://github.com/GEWIS/sudosos-backend/blob/e8cf1d272e989ca62463f2f48211c7ba5758c35c/src/gewis/gewis.ts#L43) since that is the service level of where the accounts get created.

Reply:
I think I'll park this PR for now, in order to first also work on the improved SudoSOS mailings. If you want to @JustSamuel, you can also take a look at those other things that you mentioned, as you are more familiar with the GEWISDB and LDAP integrations (and these things are difficult to test)

Reply:
> I personally don't think we should make a dependency on the GEWIS database to verify whether MEMBER accounts are actually deleted from the GEWIS database, because it seems unnecessary?

@Yoronex It is necessary to avoid one person with two accounts. In case the BAC Treasurer converts a member to a local account and later the same member wants to join a GEWIS activity or view photos on the website so now the member is activated in GEWISDB either way. They then sign in to sudosos with that account and suddenly they have two accounts.
Remember that maintaining local accounts is very undesirable for people who reslly should just have become graduates, because now suddenly the bacpm has to make sure the personal data is always updated instead of the secretary of GEWIS